---
title: "FORGENIUS-pipeline, test 20 SNPs"
author: "SCGM & MW"
date: "`r Sys.Date()`"
output: github_document
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load packages

```{r load packages, message=F, warning=F}
rm(list = ls()) # cleans the working environment

#IMPORTANT: don't change order of libraries as some mask others!
library(vcfR)
library(adegenet)
library(hierfstat)
library(LEA)
library(dplyr)
library(mapplots)
library(shapefiles)
library(miscTools)
library(dartR)
library(poppr)
library(ggplot2)

# maybe needed, to check:
#library(knitr)
#library(htmltools)
#library(stringr)
#library(kableExtra)
#library(radiant.data)


# Function to build tables
kable_mydf <- function(x, boldfirstcolumn,font_size){
  x %>% 
    kable() %>%  
    kable_styling(font_size=font_size,
                  bootstrap_options = c("stripped","hover", "condensed"), full_width = F) %>% 
    {if(boldfirstcolumn == TRUE) column_spec(., 1, bold = T) else .}
}
```

Read data, filtering and convert to different formats

```{r read data}
# keep only random SNPs in VCF file using a bed file to filter in vcftools
# system2("functions/extract_random.sh") # change input and output file names in functions/extract_random.sh

# read data from VCF into genind format
species_name = "Aalba" #change species name if needed
suffix = "_20SNPs" #change suffix if needed
data_file <- paste("data/", species_name, suffix, ".vcf.gz", sep="")
data_vcf <- read.vcfR(data_file) 
data_genind <- vcfR2genind(data_vcf)
pop(data_genind) <- substr(indNames(data_genind), 1,8) #takes pop name from the first 8 digits of sample name, e.g. AUT00215

# read data from genotype file (e.g., from Axiom chips) into genind format
#data <- read.table("data/POPGEN_data_curated_Affx_2use_JULY2022_miss30.txt", header=T)
#data[1:10,1:10] %>% kable_mydf(boldfirstcolumn = F, font_size = 11) # visualization
#data_genind <- df2genind(X = data[, -(1:2)], sep="/", pop = data[, 2], ind.names = data[, 1], NA.char ="NA") #change column numbers, separator or missing data string if needed

# Remove loci with too much missing data from genind object
#data_genind <- missingno(data_genind, type = "loci", cutoff = 0.05, quiet = FALSE, freq = FALSE)

# Remove individuals with too much missing data from genind object
#data_genind <- missingno(data_genind, type = "geno", cutoff = 0.05, quiet = FALSE, freq = FALSE)

# Filter genind by locus list
#loci2keep <- scan(file="gea_outliers.txt", what=character()) # read list of loci to keep
#data_genind = data_genind[loc=loci2keep] # filter by locus name in list

# Filter by Minor Allele Frequency (MAF) from genind object
#data_genind <- informloci(data_genind, cutoff = 2/nInd(data_genind), MAF = 0.01, quiet = FALSE)

# extract individual names
individual_names<-indNames(data_genind)

# genpop format (don't mistake wiwth genepop format, below)
data_genpop <- genind2genpop(data_genind)
print(data_genpop)
pop_names<-popNames(data_genpop)
nb_pop<-length(pop_names)

# counting individuals per population
pop_labels <- data_genind@pop
individuals_per_pop <- table(pop_labels)

# hierfstat format
data_hierfstat <- genind2hierfstat(data_genind,pop=NULL)

# bayescan format to external file
write.bayescan(dat=data_hierfstat,diploid=TRUE,fn="bayescan/data_bayescan")

# structure format to external file
write.struct(dat=data_hierfstat,ilab=individual_names,pop=NULL,MARKERNAMES=FALSE,MISSING=-9,fname="snmf/data_structure") 

# geno format to external file
struct2geno (input.file="snmf/data_structure", ploidy=2, FORMAT=2, extra.row=0, extra.column=1) #writing this file can take quite long

# genepop format to external file
data_gl <- gi2gl(data_genind) # convert first to genlight object

data_genepop <- gl2genepop(
  data_gl,
  outfile = "data_genepop",
  outpath = "./metapop",
  pop_order = "alphabetic",
  output_format = "2_digits",
  verbose = 5
) 
```

Run admixture analysis with LEA

```{r run SNMF, include=FALSE}
# run SNMF (use include=FALSE to remove annoying console output in report)
data_snmf <- snmf(input.file = "snmf/data_structure.geno",
                   K = 1:10, #indicate max number of K
                   entropy = TRUE,
                   repetitions = 5,
                   project = "new",
                   CPU = 10, #indicate the number of CPUs to use
                   alpha = 100
                  )
```

Estimate admixture coefficients

```{r Estimate admixture coefficients}
# processing sNMF results
plot(data_snmf, cex = 1.2, col = "lightblue", pch = 19) # run this line separately to determine best K
best_K = 5  #select best K based on entropy graph
ce <-  cross.entropy(data_snmf, K = best_K)
best_run <- which.min(ce)

# plot barchart for best run of best K
barchart (data_snmf, best_K, best_run, sort.by.Q = T, col = rainbow(best_K), border = NA, space = 0, xlab = "Individuals", ylab = "Admixture coefficients") #use "sort.by.Q = F, names.arg= individual_names, cex.names = 0.4, las=2, " to get ind labels in original order

# estimate ancestry coefficients by pop
q_mat <- LEA::Q(data_snmf, K = best_K, run = best_run) 
colnames(q_mat) <- paste0("Q", 1:best_K)
q_matrix <- data.frame(unlist(q_mat))
q_matrix <- q_matrix %>%
  mutate(individual_names, pop_labels)
q_by_pop <- aggregate(q_matrix[, 1:best_K], list(q_matrix$pop_labels), mean)

# clean-up
#remove.snmfProject("snmf/data_structure.snmfProject")

# plot map for best K
data_coord <- paste("data/", species_name, "_coord.txt", sep="")
coord = read.table(data_coord, header = T) #GCU, longitude and latitude in this order
Npop = length(unique(pop_names))
qmatrix2plot <- merge(coord, q_by_pop, by.x = "GCU", by.y = "Group.1")
xlim <- c(-10,30) 
ylim <- c(35,55)  #adjust if needed
basemap(xlim, ylim) 
#shape <- read.shapefile("shapefiles/chorological_maps_dataset/Abies_alba_plg") #change species name
shape <- read.shapefile("shapefiles/Europe_merged")
draw.shape(shape, col = "grey90")
for (i in 1:Npop){
  z_var=NULL
  for (j in 4:(best_K+3)){z_var <- append(z_var, qmatrix2plot[i,j])}
  add.pie(z = z_var, x = qmatrix2plot[i,2], y = qmatrix2plot[i,3], labels =   "", col = rainbow(best_K))
 }
```

Compute number of polymorphic loci per population

```{r number of polymorphic loci}

# Estimate allele frequency in adegenet
allele_freq <- tab(data_genpop)

# Count monomorphic loci and missing loci per population
monomorphic_counts <- rowSums(allele_freq == 1)
nb_loci <- rowSums(allele_freq != "NA")/2
polymoprhic_loci <- 1-(monomorphic_counts/nb_loci)
```

Compute genetic diveristy and inbreeding stats with hierfstat

```{r genetic diversity & inbreeding}
# Compute basic statistics
Hs <- Hs(data_hierfstat)
Ho <- Ho(data_hierfstat)
Fis <- 1-(Ho/Hs)
```

Compute population genetic and allelic diversity contributions with metapop2

```{r genetic and allelic diversity partitions}
# Genetic and allelic diversity partitions with metapop2
system2("./metapop/metapop", arg = "metapop/data_genepop metapop/config")

# Clean-up
file.copy("res_run_FULL.md", "metapop/results")
file.copy("res_run_p.csv", "metapop/results")
file.remove("res_run_FULL.md")
file.remove("res_run_p.csv")

# Read all lines from the file
file_path <- "metapop/results/res_run_FULL.md"
lines <- readLines(file_path)

# GENETIC DIVERSITY CONTRIBUTIONS
# Identify the start of the table
start_line <- grep("Percentage of loss \\(\\+\\) or gain \\(-\\) of gene diversity after removal of each subpopulation", lines) + 4

# Extract the lines containing the table data
table_lines <- lines[start_line:(start_line + (nb_pop-1))]

# Clean up the lines: remove leading/trailing spaces and split by '|' character
table_data <- lapply(table_lines, function(line) {
  # Remove extra spaces and split by '|'
  values <- strsplit(trimws(line), "\\|")[[1]]
  # Further clean the values by trimming and removing any extra characters
  trimws(values)
})

# Convert list of vectors to a data frame
contrib_div <- as.data.frame(do.call(rbind, table_data), stringsAsFactors = FALSE)

# Set column names appropriately
names(contrib_div) <- c("ID1", "cHs", "cDg", "cHt")

# Add pop levels
contrib_div <- contrib_div %>% mutate(GCU=pop_names, .after=ID1)

# Convert to numeric
contrib_div$cHs <- as.numeric(contrib_div$cHs)
contrib_div$cDg <- as.numeric(contrib_div$cDg)
contrib_div$cHt <- as.numeric(contrib_div$cHt)

# ALLELIC DIVERSITY CONTRIBUTIONS
# Identify the start of the table
start_line <- grep("Percentage of loss \\(\\+\\) or gain \\(-\\) of allelic diversity after removal of each subpopulation", lines) + 4

# Extract the lines containing the table data (next 3 lines)
table_lines <- lines[start_line:(start_line + (nb_pop-1))]

# Clean up the lines: remove leading/trailing spaces and split by '|' character
table_data <- lapply(table_lines, function(line) {
  # Remove extra spaces and split by '|'
  values <- strsplit(trimws(line), "\\|")[[1]]
  # Further clean the values by trimming and removing any extra characters
  trimws(values)
})

# Convert list of vectors to a data frame
contrib_a <- as.data.frame(do.call(rbind, table_data), stringsAsFactors = FALSE)

# Set column names appropriately
names(contrib_a) <- c("ID2", "cAs", "cDa", "cAt")

# Add pop levels
contrib_a <- contrib_a %>% mutate(GCU=pop_names, .after=ID2)

# Convert to numeric
contrib_a$cAs <- as.numeric(contrib_a$cAs)
contrib_a$cDa <- as.numeric(contrib_a$cDa)
contrib_a$cAt <- as.numeric(contrib_a$cAt)

# Merge dataframes
contrib <- merge(contrib_div, contrib_a, by = "GCU") # Careful, merge() reorder the populations by alphabetical order (as in the output)
contrib <- subset(contrib, select = -c(ID1, ID2))
str(contrib)

# Create contribution figures
# Genetic diversity
contrib_temp1 <- subset(contrib, select = c(GCU, cHs))
colnames(contrib_temp1) <- c("GCU", "values")
contrib_temp1 <- contrib_temp1 %>% mutate(Partition="Diversity", .after=values)
contrib_temp2 <- subset(contrib, select = c(GCU, cDg))
colnames(contrib_temp2) <- c("GCU", "values")
contrib_temp2 <- contrib_temp2 %>% mutate(Partition="Differentiation", .after=values)
contrib_fig_div <- rbind(contrib_temp1,contrib_temp2)
ggplot(contrib_fig_div, aes(x = GCU, y = values))+ geom_col(aes(fill = Partition), width = 0.7) + scale_y_continuous() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + xlab("GCUs") + ylab("Contribution % (genetic diversity)") + scale_alpha_manual(values = 1) + labs(alpha="")

# Allelic diversity
contrib_temp3 <- subset(contrib, select = c(GCU, cAs))
colnames(contrib_temp3) <- c("GCU", "values")
contrib_temp3 <- contrib_temp3 %>% mutate(Partition="Diversity", .after=values)
contrib_temp4 <- subset(contrib, select = c(GCU, cDa))
colnames(contrib_temp4) <- c("GCU", "values")
contrib_temp4 <- contrib_temp4 %>% mutate(Partition="Differentiation", .after=values)
contrib_fig_allelic <- rbind(contrib_temp3,contrib_temp4)
ggplot(contrib_fig_allelic, aes(x = GCU, y = values))+ geom_col(aes(fill = Partition), width = 0.7) + scale_y_continuous() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + xlab("GCUs") + ylab("Contribution % (allelic diversity)") + scale_alpha_manual(values = 1) + labs(alpha="")
```

Compute genetic distinctness

```{r genetic differentiation}
# Average pairwise Fst
pairwise_WCfst <- pairwise.WCfst(data_hierfstat) #this is very slow, took about 16 hours for 15 pops and 90k SNPs in a laptop
ave_pairwise_fst<-rowMeans(pairwise_WCfst, na.rm = TRUE) 

# Population-specific Fst from bayescan

#If you are running RStudio in Windows 10
#system2("powershell", arg = c("-file", "bayescan.ps1")) #check out running parameters in ps1 file; bayescan is very slow in Windows, took about 5 hours for 15 pops and 14k SNPs using 12 CPUs and standard parameters

#If you are running RStudio in Linux
system2("./bayescan/BayeScan2.1_linux64bits", arg = "bayescan/data_bayescan -n 5000 -thin 10 -nbp 10 -pilot 5000 -burn 10000 -pr_odds 1000 -od bayescan/results")

source ("bayescan/plot_R.r")
plot_bayescan("bayescan/results/data_bayescan_fst.txt",add_text=T,pos=0.02,size=0.7,FDR=0.05)
output_bayescan<-read.table("bayescan/results/data_bayescan.sel")
output_bayescan$logL<-NULL
pop_specific_fst<-colMeans(output_bayescan, na.rm = TRUE)
```

Print table

```{r print table}
# prepare dataframe with basic stats
table_basic_stats<-data.frame(unlist(individuals_per_pop), unlist(nb_loci), unlist(polymoprhic_loci), unlist(Hs), unlist(Fis), unlist(ave_pairwise_fst), unlist(pop_specific_fst), row.names = NULL)
names (table_basic_stats) = c("GCU", "N", "nb_loci", "polymorphic_loci", "Hs", "Fis", "ave_pairwise_fst", "pop_specific_fst")

# merge dataframes in a single table
table_results <- merge(table_basic_stats, q_by_pop, by.x = "GCU", by.y = "Group.1")
table_results_contrib <- merge(table_results, contrib, by = "GCU")
# merge dataframes by GCU name, it orders the new dataframe alphabetically
data_print <- paste("results/table_results_contrib_", species_name, suffix, ".csv", sep="")
write.csv(table_results_contrib, data_print)
```
