---
title: "FORGENIUS-pipeline"
author: "SCGM & MW"
date: "`r Sys.Date()`"
output: github_document
---
```{r load packages}
#IMPORTANT: don't change order of libraries as some mask others!
library(vcfR)
library(adegenet)
#library(genepop)
library(graph4lg)
library(hierfstat)
#library(VariantAnnotation)
#library(genomation)
#library(pegas)
#library(GenomicRanges)
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Read data and convert to different formats

```{r read data}
data_vcf <- read.vcfR("data/Aalba_target.vcf.gz") #change species name

# subset VCF files based on bed file - do it in vcftools?
#bed_gr <- readBed("data/test_bed")
#snp_positions <- GRanges(seqnames = data_vcf@fix$CHROM,                     ranges = IRanges(start = data_vcf@fix$POS, width = 1))
# overlaps <- findOverlaps(snp_positions, bed)
# vcf_filtered <- vcf[queryHits(overlaps),]
#data_vcf_random <- subsetByOverlaps(data_vcf, bed_gr)
#writeVcf(data_vcf_random, "data/subsetted_file.vcf")

# adegenet formats (genind and genpop)
data_genind <- vcfR2genind(data_vcf)
pop(data_genind) <- substr(indNames(data_genind), 1,8) #takes pop name from the first 8 digits of sample name, e.g. AUT00215
data_genpop <- genind2genpop(data_genind)
print(data_genpop)
pop_names<-popNames(data_genpop)

# counting individuals per population
pop_labels <- data_genind@pop
individuals_per_pop <- table(pop_labels)

# hierfstat format
data_hierfstat <- genind2hierfstat(data_genind,pop=NULL)

# bayescan format
#write.bayescan(dat=data_hierfstat,diploid=TRUE,fn="data.bayescan")

# genepop format
#genind_to_genepop(data_genind, output = "file.txt")

```

Compute number of polymorphic loci per population

```{r number of polymorphic loci}

# Estimate allele frequency in adegenet
allele_freq <- tab(data_genpop)

# Count monomorphic loci and missing loci per population
monomorphic_counts <- rowSums(allele_freq == 1)
nb_loci <- rowSums(allele_freq != "NA")/2
polymoprhic_loci <- 1-(monomorphic_counts/nb_loci)
```

Compute basic stats with hierfstat

```{r genetic diversity & inbreeding}
Hs <- Hs(data_hierfstat)
Ho <- Ho(data_hierfstat)
Fis <- 1-(Ho/Hs)
```

Compute average pairwise genetic differentiation with hierfstat

```{r genetic differentiation}
pairwise_WCfst <- pairwise.WCfst(data_hierfstat)
fst_pop<-rowMeans(pairwise_WCfst, na.rm = TRUE)
```

Print table

```{r print table}
table_results<-data.frame(unlist(nb_loci), unlist(individuals_per_pop), unlist(polymoprhic_loci), unlist(Hs), unlist(Fis), unlist(fst_pop))
names (table_results) = c("nb_loci", "GCU", "N", "polymorphic_loci", "Hs", "Fis", "Fst-pop")
write.csv(table_results,"table_results_Aalba_target.csv") #change species name
```
