---
title: "FORGENIUS-pipeline"
author: "SCGM & MW"
date: "`r Sys.Date()`"
output: github_document
---
```{r load packages}
#IMPORTANT: don't change order of libraries as some mask others!
library(vcfR)
library(adegenet)
#library(genepop)
library(graph4lg)
library(hierfstat)
#library(radiator)
#library(VariantAnnotation)
#library(genomation)
#library(pegas)
#library(GenomicRanges)
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Read data and convert to different formats

```{r read data}
data_vcf <- read.vcfR("data/Msylvestris_SNPs_filt.vcf.gz") #change species name

# subset VCF files based on bed file - do it in vcftools!

# adegenet formats (genind and genpop)
data_genind <- vcfR2genind(data_vcf)
pop(data_genind) <- substr(indNames(data_genind), 1,8) #takes pop name from the first 8 digits of sample name, e.g. AUT00215
data_genpop <- genind2genpop(data_genind)
print(data_genpop)
pop_names<-popNames(data_genpop)

# counting individuals per population
pop_labels <- data_genind@pop
individuals_per_pop <- table(pop_labels)

# hierfstat format
data_hierfstat <- genind2hierfstat(data_genind,pop=NULL)

# bayescan format to external file
write.bayescan(dat=data_hierfstat,diploid=TRUE,fn="data_bayescan") #writting this file can take about 6 hours for 90k SNPs

# genepop format to external file
#genind_to_genepop(data_genind, output = "data_genepop.txt")
```

Compute number of polymorphic loci per population

```{r number of polymorphic loci}

# Estimate allele frequency in adegenet
allele_freq <- tab(data_genpop)

# Count monomorphic loci and missing loci per population
monomorphic_counts <- rowSums(allele_freq == 1)
nb_loci <- rowSums(allele_freq != "NA")/2
polymoprhic_loci <- 1-(monomorphic_counts/nb_loci)
```

Compute basic stats with hierfstat

```{r genetic diversity & inbreeding}
Hs <- Hs(data_hierfstat)
Ho <- Ho(data_hierfstat)
Fis <- 1-(Ho/Hs)
```

Compute genetic distinctness

```{r genetic differentiation}
# Average pairwise Fst
pairwise_WCfst <- pairwise.WCfst(data_hierfstat) #this is very slow, took about 16 hours for 15 pops and 90k SNPs in a laptop
ave_pairwise_fst<-rowMeans(pairwise_WCfst, na.rm = TRUE) 

# Population-specific Fst from bayescan

#If you are running RStudio in Windows 10
system2("powershell", arg = c("-file", "bayescan.ps1")) #check out running parameters in ps1 file; bayescan is very slow, took about xxx hours for 15 pops and 90k SNPs using 12 CPUs and standard parameters
source ("bayescan/plot_R.r")
plot_bayescan("data_bayescan_fst.txt",add_text=T,pos=0.02,size=0.7,FDR=0.05)
output_bayescan<-read.table("data_bayescan.sel")
output_bayescan$logL<-NULL
pop_specific_fst<-colMeans(output_bayescan, na.rm = TRUE)

#If you are running RStudio in Linux - not tested yet!
#scan.pops <- radiator::run_bayescan(
#    data = "data_bayescan",
#    n = 5000,
#    thin = 10,
#    nbp = 20,
#    pilot = 5000,
#    burn = 50000,
#    subsample = NULL,
#    iteration.subsample = 1,
#    parallel.core = parallel::detectCores() - 1,
#    pr_odds = 1000,
#    bayescan.path = "bayescan/BayeScan2.1_linux64bits"
#    )
```

Print table

```{r print table}
table_results<-data.frame(unlist(individuals_per_pop), unlist(nb_loci), unlist(polymoprhic_loci), unlist(Hs), unlist(Fis), unlist(ave_pairwise_fst), unlist(pop_specific_fst), row.names = NULL)
names (table_results) = c("GCU", "N", "nb_loci", "polymorphic_loci", "Hs", "Fis", "ave_pairwise_fst", "pop_specific_fst")
write.csv(table_results,"table_results_Msylvestris.csv") #change species name
```
